.github/workflows/continuous-integration.yml

```diff
@@ -0,0 +1,108 @@
+name: continuous-integration
+on: [push]
+jobs:
```

Comment:
Once we get this running we can look at relative runtimes of each test suite (unit, functional, integration, system) and try breaking them into parallel jobs to see if we can get the whole thing to run faster.